app/models/way.rb

```diff
@@ -248,6 +300,55 @@ def delete_with_history!(new_way, user)
     end
   end
 
+  def self.delete_with_history_bulk!(ways, changeset, if_unused = false)
+    way_hash = ways.collect { |way| [way.id, way] }.to_h
+    skipped = {}
+    # need to start the transaction here, so that the database can
+    # provide repeatable reads for the used-by checks. this means it
+    # shouldn't be possible to get race conditions.
+    Way.transaction do
+      way_ids = way_hash.keys
+      old_ways = Way.select("id, version, visible").where(:id => way_ids).lock
+      raise OSM::APIBadUserInput, "Way not exist. id: " + (way_ids - old_ways.collect(&:id)).join(", ") unless way_ids.length == old_ways.length
```

Comment:
Non-existing object should return HTTP 404 (Not found) instead of HTTP 400 (Bad Request)